cmd/gateway-main.go

```diff
@@ -233,6 +234,9 @@ func StartGateway(ctx *cli.Context, gw Gateway) {
 		globalHTTPServer.Shutdown()
 		logger.FatalIf(err, "Unable to initialize gateway backend")
 	}
+	if globalAutoEncryption && !newObject.IsEncryptionSupported() {
+		logger.Fatal(errors.New("Invalid KMS configuration"), "auto-encryption is enabled but backend does not support encryption")
+	}
```

Comment:
@aead, this would be a misleading error message - backend could support encryption but it may be not be turned on for the gateway in question.

Reply:
AFAIK it is not configurable whether the backend supports encryption?!
At the moment only FS & XL support encryption and we will enable it for AWS soon. All other backends don't support encryption (in our code).

Reply:
That's true @aead, but as a UI message it is misleading because the backend could support encryption (example s3), but our gateway has not enabled it.
maybe ``` "auto-encryption is enabled but gateway does not support encryption"```

Reply:
Okay, changing the error message :+1:

Reply:
Oh yeah, config was not load when check happened. Fixed